var/spack/repos/builtin/packages/glvis/package.py

```diff
@@ -0,0 +1,131 @@
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class Glvis(Package):
+    """GLVis: an OpenGL tool for visualization of FEM meshes and functions"""
+
+    homepage = "http://glvis.org/"
+    url      = "https://github.com/GLvis/glvis/archive/v3.3.tar.gz"
+
+    # glvis (like mfem) is downloaded from a URL shortener at request
+    # of upstream author Tzanio Kolev <tzanio@llnl.gov>.  See here:
+    # https://github.com/mfem/mfem/issues/53
+    #
+    # The following procedure should be used to verify security when a
+    # new version is added:
+    #
+    # 1. Verify that no checksums on old versions have changed.
+    #
+    # 2. Verify that the shortened URL for the new version is listed at:
+    #    http://glvis.org/download/
+    #
+    # 3. Use http://getlinkinfo.com or similar to verify that the
+    #    underling download link for the latest version comes has the
+    #    prefix: http://glvis.github.io/releases
+    #
+    # If this quick verification procedure fails, additional discussion
+    # will be required to verify the new version.
+
+    version('3.3',
+            '1201a76d1b0c38240186c06f6478de77',
+            url='http://goo.gl/C0Oadw',
+            extension='.tar.gz')
+
+    version('3.2',
+            '4575a03a50d4730c07b5d4018e936707',
+            url='http://goo.gl/hzupg1',
+            extension='.tar.gz')
+
+    version('3.1',
+            '3ce8a53209c6593e066e568dbbf2bbf6',
+            url='http://goo.gl/gQZuu9',
+            extension='tar.gz')
+
+    variant('png', default=True,
+            description='Build with libpng support for screenshots')
+    variant('tiff', default=True,
+            description='Build with libtiff support for screenshots')
+    variant('fonts', default=True,
+            description='Build with font support via freetype & fontconfig')
+
+    depends_on('mfem@3.3:3.3.999', when='@3.3')
```

Comment:
The version of mfem should be fixed at `3.3` when building glvis `3.3`. The newer versions of mfem (e.g. `3.3.2`) will not work properly with glvis `3.3`.

Also, to support the older versions, should there be lines like this for `3.2` and `3.1`?

Reply:
Yes. Added these.